### PR TITLE
[DC-2247] Replace bq.get_client(...) calls in data_steward/tools

### PR DESCRIPTION
**Pull request by lledezma**
Implements the new `BigQueryClient` by replacing `bq.get_client(...)` calls in data_steward/tools folder files:
- data_steward/tools/add_hpo.py
- data_steward/tools/clean_project_datasets.py
- data_steward/tools/copy_dataset_to_output_prod.py
- data_steward/tools/recreate_person.py
- data_steward/tools/snapshot_by_query_test.py
- Associated tests

### CODE REVIEW

File: data_steward/tools/copy_dataset_to_output_prod.py [hunk 1]
```diff
@@ -145,17 +146,17 @@ def get_arg_parser() -> argparse.ArgumentParser:
...
         f'Copying tables from dataset {args.src_project_id}.{args.src_dataset_id} to {args.output_prod_project_id}.{output_dataset_name}...'
     )
-    bq.copy_datasets(client, f'{args.src_project_id}.{args.src_dataset_id}',
+    bq.copy_datasets(bq_client, f'{args.src_project_id}.{args.src_dataset_id}',
                      f'{args.output_prod_project_id}.{output_dataset_name}')
 
     #Append extra columns to person table
     LOGGER.info(f'Appending extract columns to the person table...')
-    update_person(client, args.output_prod_project_id, output_dataset_name)
+    update_person(bq_client, output_dataset_name)
```
Comment (nishanthpp93) [inline data_steward/tools/copy_dataset_to_output_prod.py:160]: This is good to highlight. We don't have to rely on the project being passed as long as the client is instantiated with it.

File: data_steward/tools/snapshot_by_query.py [hunk 2]
```diff
@@ -115,21 +116,21 @@ def get_copy_table_query(project_id, dataset_id, table_id, client):
...
 
 def copy_tables_to_new_dataset(project_id, dataset_id, snapshot_dataset_id):
     """
     lists the tables in the dataset and copies each table to a new dataset.
-    :param dataset_id:
-    :param project_id:
-    :param snapshot_dataset_id:
+    :param dataset_id: identifies the source dataset
+    :param project_id: identifies the project
+    :param snapshot_dataset_id: identifies the destination dataset
```
Comment (nishanthpp93) [inline data_steward/tools/snapshot_by_query.py:129]: 👍

**Review by nishanthpp93 — COMMENTED** [review]
Overall looks good! Thanks for updating the docstrings as well. Added a suggestion for you to try out, let me know your thoughts.

File: tests/unit_tests/data_steward/tools/snapshot_by_query_test.py [hunk 3]
```diff
@@ -17,7 +17,9 @@ def setUpClass(cls):
         print(cls.__name__)
         print('**************************************************************')
 
-        cls.mock_client = ''
+        cls.mock_bq_client = mock.MagicMock()
+        type(cls.mock_bq_client).project = mock.PropertyMock(
+            return_value='test-project')
```
[the remark this answers is not in the document]
Comment (lledezma) [reply tests/unit_tests/data_steward/tools/snapshot_by_query_test.py:22]: This works too.

**Review by nishanthpp93 — APPROVED** [review]
Looks good!

**Review by dcarbone — APPROVED** [review]
lgtm :+1:

**Pull request merged**